src/components/Main/Results/DeterministicLinePlot.tsx

```diff
@@ -220,23 +220,9 @@ export function DeterministicLinePlot({
   const tMin = _.minBy(plotData, 'time')!.time // eslint-disable-line @typescript-eslint/no-non-null-assertion
   const tMax = _.maxBy(plotData, 'time')!.time // eslint-disable-line @typescript-eslint/no-non-null-assertion
 
-  const reducedObservationsToPlot = translatePlots(t, observationsToPlot(caseTimeWindow)).filter((itemToPlot) => {
+  const observationsHavingDataToPlot = observationsToPlot(caseTimeWindow).filter((itemToPlot) => {
     if (observations.length !== 0) {
-      if (countObservations.cases && itemToPlot.key === DATA_POINTS.ObservedCases) {
-        return true
-      }
-      if (countObservations.newCases && itemToPlot.key === DATA_POINTS.ObservedNewCases) {
-        return true
-      }
-      if (countObservations.hospitalized && itemToPlot.key === DATA_POINTS.ObservedHospitalized) {
-        return true
-      }
-      if (countObservations.ICU && itemToPlot.key === DATA_POINTS.ObservedICU) {
-        return true
-      }
-      if (countObservations.observedDeaths && itemToPlot.key === DATA_POINTS.ObservedDeaths) {
-        return true
-      }
+      return hasObservations[itemToPlot.key]
```

Comment:
We can discover presence according to a DATA_POINTS label, so the list of conditional statements can be replaced by a single lookup.